### ISSUE TITLE
Daemon crashes on start with "undefined symbol" error

### ISSUE THREAD

**Issue by marxjohnson** [Librevault/librevault#67]
On Ubuntu 16.04, installed from the repository:
<pre>
mark@coachz:~$ librevault-daemon 
   __    __ _                                _ __ 
  / /   /_/ /_  ____ _____ _  __ ___  __  __/ / /_
 / /   __/ /_ \/ ___/ ___/ / / / __ \/ / / / / __/
/ /___/ / /_/ / /  / ___/\ \/ / /_/ / /_/ / / /___
\____/_/\____/_/  /____/  \__/_/ /_/\____/_/\____/
[2016-09-10 10:57:10.156566] [T:13094] [I] Librevault 0.1.18.7
[2016-09-10 10:57:10.159266] [T:13094] [I] [WSService] Listening on [::]:42345
[2016-09-10 10:57:10.159386] [T:13094] [I] [MulticastDiscovery] Started Local Peer Discovery on: 239.192.152.144:28914
[2016-09-10 10:57:10.159464] [T:13094] [I] [MulticastDiscovery] Started Local Peer Discovery on: [ff08::bd02]:28914
[2016-09-10 10:57:10.159625] [T:13094] [N] Loading 61 nodes from session file
[CONTROL] Librevault Control is listening at ws://[::]:42346
[2016-09-10 10:57:10.162513] [T:13094] [I] [pool:bulk_ios] Threads: 4
[2016-09-10 10:57:10.163125] [T:13094] [I] [pool:network_ios] Threads: 1
[2016-09-10 10:57:10.163236] [T:13094] [I] [pool:etc_ios] Threads: 1
librevault-daemon: symbol lookup error: librevault-daemon: undefined symbol: _ZN6google8protobuf2io17CodedOutputStream12VarintSize64Em
</pre>
I can launch the gui, but adding a folder doesn't work:
<pre>
mark@coachz:~$ librevault-gui
"   __    __ _                                _ __ \n"
"  / /   /_/ /_  ____ _____ _  __ ___  __  __/ / /_\n"
" / /   __/ /_ \\/ ___/ ___/ / / / __ \\/ / / / / __/\n"
"/ /___/ / /_/ / /  / ___/\\ \\/ / /_/ / /_/ / / /___\n"
"\\____/_/\\____/_/  /____/  \\__/_/ /_/\\____/_/\\____/\n"
"[CONTROL] Librevault Control is listening at ws://[::]:42346\n"
Connecting to:  QUrl("ws://localhost:42346")
Connecting to daemon:  QUrl("ws://localhost:42346")
Disconnected from daemon:  QUrl("ws://localhost:42346")  Reason:  ""
</pre>

**Comment by GamePad64** [comment]
This is really strange. Looks like there is something wrong with Protobuf installation. Can you please post here the output of these commands:
```
apt-cache policy libprotobuf10
```

```
objdump -x /usr/lib/*/libprotobuf* | grep _ZN6google8protobuf2io17CodedOutputStream12VarintSize64Em
```

```
ldd /usr/bin/librevault-daemon
```

**Comment by marxjohnson** [comment]
I've just upgraded to 0.1.18.10 and I am still experiencing the problem.  The output you asked for:
<pre>
mark@coachz:~$ apt-cache policy libprotobuf10
libprotobuf10:
  Installed: 3.0.0~beta2-0~0ubuntu16.04
  Candidate: 3.0.0-6~ubuntu16.04
  Version table:
     3.0.0-6~ubuntu16.04 500
        500 https://releases.librevault.com/debian xenial/main amd64 Packages
 *** 3.0.0~beta2-0~0ubuntu16.04 100
        100 /var/lib/dpkg/status
</pre>
<pre>
mark@coachz:~$ objdump -x /usr/lib/*/libprotobuf* | grep _ZN6google8protobuf2io17CodedOutputStream12VarintSize64Em
</pre>
<pre>mark@coachz:~$ ldd /usr/bin/librevault-daemon
	linux-vdso.so.1 =>  (0x00007fff1e870000)
	libcrypto++.so.6 => /usr/lib/x86_64-linux-gnu/libcrypto++.so.6 (0x00007f15fb186000)
	libboost_system.so.1.58.0 => /usr/lib/x86_64-linux-gnu/libboost_system.so.1.58.0 (0x00007f15faf82000)
	libboost_filesystem.so.1.58.0 => /usr/lib/x86_64-linux-gnu/libboost_filesystem.so.1.58.0 (0x00007f15fad69000)
	libboost_iostreams.so.1.58.0 => /usr/lib/x86_64-linux-gnu/libboost_iostreams.so.1.58.0 (0x00007f15fab50000)
	libboost_locale.so.1.58.0 => /usr/lib/x86_64-linux-gnu/libboost_locale.so.1.58.0 (0x00007f15fa868000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f15fa64a000)
	libnatpmp.so.1 => /usr/lib/x86_64-linux-gnu/libnatpmp.so.1 (0x00007f15fa447000)
	libsqlite3.so.0 => /usr/lib/x86_64-linux-gnu/libsqlite3.so.0 (0x00007f15fa172000)
	libprotobuf.so.10 => /usr/lib/x86_64-linux-gnu/libprotobuf.so.10 (0x00007f15f9d2b000)
	libssl.so.1.0.0 => /lib/x86_64-linux-gnu/libssl.so.1.0.0 (0x00007f15f9ac2000)
	libcrypto.so.1.0.0 => /lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007f15f9667000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f15f92e4000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f15f8fdb000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f15f8dc5000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f15f89fb000)
	libbz2.so.1.0 => /lib/x86_64-linux-gnu/libbz2.so.1.0 (0x00007f15f87eb000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f15f85d1000)
	libboost_thread.so.1.58.0 => /usr/lib/x86_64-linux-gnu/libboost_thread.so.1.58.0 (0x00007f15f83aa000)
	libicui18n.so.55 => /usr/lib/x86_64-linux-gnu/libicui18n.so.55 (0x00007f15f7f48000)
	libicuuc.so.55 => /usr/lib/x86_64-linux-gnu/libicuuc.so.55 (0x00007f15f7bb4000)
	/lib64/ld-linux-x86-64.so.2 (0x000055f94162f000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f15f79af000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f15f77a7000)
	libicudata.so.55 => /usr/lib/x86_64-linux-gnu/libicudata.so.55 (0x00007f15f5cef000)
</pre>

**Comment by GamePad64** [comment]
Your version of `libprotobuf10` is outdated. Google have released stable version Protocol Buffers 3.0, and I started linking with it. It is binary-incompatible with beta version, so you have to install a new version.

Please, run `sudo apt-get install libprotobuf10` to do it.

**Comment by marxjohnson** [comment]
Thanks, that was successful!
